broker.go

```diff
@@ -1013,6 +1027,16 @@ func (b *Broker) sendInternal(rb protocolBody, promise *responsePromise) error {
 		return ErrUnsupportedVersion
 	}
 
+	// validate the request is using a supported API version
+	if b.brokerAPIVersions != nil && b.brokerAPIVersions[rb.key()] != [2]int16{0, 0} {
+		minMax := b.brokerAPIVersions[rb.key()]
+		if rb.version() < minMax[0] || rb.version() > minMax[1] {
+			return fmt.Errorf("%w: unsupported API version %d for %s, supported versions are [%d-%d]",
+				// protocolBody is a *sarama.XXXRequest
+				ErrUnsupportedVersion, rb.version(), reflect.TypeOf(rb).Elem().Name(), minMax[0], minMax[1])
```

Comment:
There is no need to import `reflect` just to get a type name, as `fmt.Printf` provides a specific verb for this: `%T`.
```suggestion
			return fmt.Errorf("%w: unsupported API version %d for %T, supported versions are [%d-%d]",
				// protocolBody is a *sarama.XXXRequest
				ErrUnsupportedVersion, rb.version(), rb, minMax.MinVersion, minMax.MaxVersion)
```

Reply:
TIL! Thanks 🙂